CardsCollectionViewVerticleLayout.swift

```diff
@@ -69,51 +75,51 @@ open class CardsCollectionViewLayout: UICollectionViewLayout {
                                                   deltaOffset: CGFloat(deltaOffset),
                                                   percentageDeltaOffset: percentageDeltaOffset)
         }
-       
+        
         return attributes
     }
-   
+    
     override open func layoutAttributesForItem(at indexPath: IndexPath) -> UICollectionViewLayoutAttributes? {
         let contentCenterX = collectionView.contentOffset.x + (collectionView.bounds.width / 2.0)
         let minVisibleIndex = Int(collectionView.contentOffset.x) / Int(collectionView.bounds.width)
-       
+        
         //Converted from contentOffset.x to contentOffset.y
         //and from collectionView.bounds.widt to collectionView.bounds.heigh
         let deltaOffset = Int(collectionView.contentOffset.y) % Int(collectionView.bounds.height)
         let percentageDeltaOffset = CGFloat(deltaOffset) / collectionView.bounds.height
-       
+        
         return computeLayoutAttributesForItem(indexPath: indexPath,
                                               minVisibleIndex: minVisibleIndex,
                                               contentCenterX: contentCenterX,
                                               deltaOffset: CGFloat(deltaOffset),
                                               percentageDeltaOffset: percentageDeltaOffset)
     }
-   
+    
     override open func shouldInvalidateLayout(forBoundsChange newBounds: CGRect) -> Bool {
         return true
     }
```

Comment:
public func moveToPage(position:Int){
        let totalPage = collectionView.contentOffset.x/collectionView.frame.width// getting totalNumberOfPage
        if position < Int(totalPage){
            scrollToPage(page: position, animated: true)
        }
       }
     public  func moveToNextPage()->Int{
            let pageIndex = round(collectionView.contentOffset.x/collectionView.frame.width)
            let pageNo = Int(pageIndex+1)
            scrollToPage(page: pageNo, animated: true)
            return pageNo
     }
     public  func moveToPreviousPage()->Int{
            let pageIndex = round(collectionView.contentOffset.x/collectionView.frame.width)
            let pageNo = Int(pageIndex-1)
            scrollToPage(page: pageNo, animated: true)
            return pageNo
    }
           
        
    func scrollToPage(page: Int, animated: Bool) {
        var frame: CGRect = self.collectionView.frame
        frame.origin.x = frame.size.width * CGFloat(page)
        frame.origin.y = 0
        self.collectionView.scrollRectToVisible(frame, animated: animated)
    }

Reply:
///this facilitate the swipe the card left and right programmtically